### PR TITLE
Merge UserLike use case into main

### PR DESCRIPTION
**Pull request by czz0120**
The basic architecture is done, lack of screens and UI design.

### CODE REVIEW

File: src/main/java/entities/User.java [hunk 1]
```diff
@@ -81,6 +79,9 @@ private List<Object> extractReviewBody(Review review){
         return revBody;
```
Comment (Tristal25) [inline src/main/java/entities/User.java:79]: I wonder if you can put extractReviewBody method into Reviews class

File: src/main/java/entities/User.java [hunk 2]
```diff
@@ -43,9 +43,7 @@ public User(String password, String accountName, Profile profile){
 
     public Profile displayProfile(){ return this.profile;}
 
-    public List<User> showBlocked(){ return this.blocked;}
-
-    public List<User> showLiked(){ return this.liked;}
+    public List<String> showLiked(){ return this.liked;}
```
Comment (Tristal25) [inline src/main/java/entities/User.java:46]: .toSeconds() method is not working sometimes, you can use getSeconds() method instead.

File: src/main/java/entities/User.java [hunk 3]
```diff
@@ -81,6 +79,9 @@ private List<Object> extractReviewBody(Review review){
         return revBody;
     }
 
+    public void like(String accName){
+        this.liked.add(accName);
+    }
```
Comment (Tristal25) [inline src/main/java/entities/User.java:84]: Maybe add a method to extract a list of all Users class using the list of usernames

**Review by Tristal25 — APPROVED** [review]
I added some comments for some suggestions. For the Gateway, I will create a gateway for each use cases, and you can see the methods in my pull request. Please match the method names and choose the methods that you may use to include in your inference. Tell me if you need more of them.

**Review by clarahqy — DISMISSED** [review]
looks good to me

File: src/main/java/entities/User.java (hunk 1 above)
Comment (czz0120) [reply src/main/java/entities/User.java:79]: Sorry, it is not written by me. I have no idea about the  Review use case.

**Review by clarahqy — APPROVED** [review]
conflicts need to be resolved first

**Review by Tristal25 — APPROVED** [review]
Looks nice

**Review by ryancys1234 — APPROVED** [review]
Looks very clean!

**Review by yifeiiZhang — APPROVED** [review]

**Pull request merged**